experiment/template/eVOLVER.py

```diff
@@ -0,0 +1,629 @@
+                text_file = open(file_path, "w")
+                text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                    EXP_NAME, x, time.strftime("%c")))
+                text_file.close()
+
+                # make temperature data file
+                file_name = "vial{0}_temp.txt".format(x)
+                file_path = os.path.join(EXP_DIR, 'temp', file_name)
+                text_file = open(file_path, "w").close()
+
+                # make temperature configuration file
+                file_name =  "vial{0}_tempconfig.txt".format(x)
+                file_path = os.path.join(EXP_DIR, 'temp_config', file_name)
+                text_file = open(file_path, "w")
+                text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                    EXP_NAME, x, time.strftime("%c")))
+                #initialize based on custom_script.choose_setup()
+                text_file.write("0,{0}\n".format(TEMP_INITIAL[x]))
+                text_file.close()
+
+                if OPERATION_MODE == 'turbidostat':
+                    # make pump log file
+                    file_name =  "vial{0}_pump_log.txt".format(x)
+                    file_path = os.path.join(EXP_DIR, 'pump_log', file_name)
+                    text_file = open(file_path, "w")
+                    text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                        EXP_NAME, x, time.strftime("%c")))
+                    text_file.write("0,0\n")
+                    text_file.close()
+
+                    # make ODset file
+                    file_name =  "vial{0}_ODset.txt".format(x)
+                    file_path = os.path.join(EXP_DIR, 'ODset', file_name)
+                    text_file = open(file_path, "w")
+                    text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                        EXP_NAME, x, time.strftime("%c")))
+                    text_file.write("0,0\n")
+                    text_file.close()
+
+                    # make growth rate file
+                    file_name =  "vial{0}_gr.txt".format(x)
+                    file_path = os.path.join(EXP_DIR, 'growthrate', file_name)
+                    text_file = open(file_path, "w")
+                    text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                        EXP_NAME, x, time.strftime("%c")))
+                    text_file.write("0,0\n") #initialize to 0
+                    text_file.close()
+
+                elif OPERATION_MODE == 'chemostat':
+                    #make chemostat file
+                    file_name =  "vial{0}_chemoconfig.txt".format(x)
+                    file_path = os.path.join(EXP_DIR, 'chemo_config', file_name)
+                    text_file = open(file_path, "w")
+                    text_file.write("0,0,0\n") #header
+                    text_file.write("0,0,0\n") #initialize to 0
+                    text_file.close()
+
+                else:
```

Comment:
You should probably just always make the pump log. If no mode is specified, don't make those other directories. Maybe let's discuss this with @cpmancuso. You also could just make all of the files even if they aren't used in all cases, which gives some flexibility if the user codes something a little weird, and would clean this area of the code up.

Reply:
Yeah, that makes sense